From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 15 Mar 2024 16:15:50 -0500
Subject: [PATCH] Adapt KIO service menu to Plasma 6

This is still broken as KF6 doesn't seem to support multiple globs at the same time, and I have no idea why having `application/x-flash-video` breaks it. However, it is slightly _less_ broken than before where it didn't work _at all_.

---
 Project/GNU/GUI/Makefile.am                | 6 +++---
 Project/GNU/GUI/mediainfo-gui.kde4.desktop | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Project/GNU/GUI/Makefile.am b/Project/GNU/GUI/Makefile.am
index 20a7ec255..29a547a49 100644
--- a/Project/GNU/GUI/Makefile.am
+++ b/Project/GNU/GUI/Makefile.am
@@ -38,9 +38,9 @@ install-data-hook:
 	install -dm 755 $(DESTDIR)$(datadir)/kde4/services/ServiceMenus && \
 	install -m 644 mediainfo-gui.kde4.desktop \
 	               $(DESTDIR)$(datadir)/kde4/services/ServiceMenus/mediainfo-gui.desktop && \
-	install -dm 755 $(DESTDIR)$(datadir)/kservices5/ServiceMenus && \
+	install -dm 755 $(DESTDIR)$(datadir)/kio/servicemenus && \
 	install -m 644 mediainfo-gui.kde4.desktop \
-	               $(DESTDIR)$(datadir)/kservices5/ServiceMenus/mediainfo-gui.desktop && \
+	               $(DESTDIR)$(datadir)/kio/servicemenus/mediainfo-gui.desktop && \
 	install -dm 755 $(DESTDIR)$(datadir)/icons/hicolor/256x256/apps && \
 	install -m 644 ../../../Source/Resource/Image/MediaInfo.png \
 	               $(DESTDIR)$(datadir)/icons/hicolor/256x256/apps/mediainfo.png && \
@@ -54,7 +54,7 @@ install-data-hook:
 uninstall-local:
 	( cd '$(DESTDIR)$(datadir)/apps/konqueror/servicemenus' && rm -f mediainfo-gui.desktop ) && \
 	( cd '$(DESTDIR)$(datadir)/kde4/services/ServiceMenus' && rm -f mediainfo-gui.desktop ) && \
-	( cd '$(DESTDIR)$(datadir)/kservices5/ServiceMenus' && rm -f mediainfo-gui.desktop ) && \
+	( cd '$(DESTDIR)$(datadir)/kio/servicemenus' && rm -f mediainfo-gui.desktop ) && \
 	( cd '$(DESTDIR)$(datadir)/icons/hicolor/256x256/apps' && rm -f mediainfo.png ) && \
 	( cd '$(DESTDIR)$(datadir)/icons/hicolor/scalable/apps' && rm -f mediainfo.svg ) && \
 	( cd '$(DESTDIR)$(datadir)/pixmaps' && rm -f mediainfo.xpm )
diff --git a/Project/GNU/GUI/mediainfo-gui.kde4.desktop b/Project/GNU/GUI/mediainfo-gui.kde4.desktop
index 695cc319d..046843679 100644
--- a/Project/GNU/GUI/mediainfo-gui.kde4.desktop
+++ b/Project/GNU/GUI/mediainfo-gui.kde4.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Service
-Name=mediainfo-gui.desktop
-ServiceTypes=KonqPopupMenu/Plugin,audio/*,video/*,image/*,application/x-flash-video
+ServiceTypes=KonqPopupMenu/Plugin;
+MimeType=audio/*,video/*,image/*;
 Actions=mediainfo-gui
 X-KDE-Priority=TopLevel
 Keywords=media info;metadata;tag;video;audio;subtitles;chapters;codec;mkv;mpeg;mp3;h264;avc;h265;hevc;avi;
